workers/data_refinery_workers/processors/management/commands/create_compendia.py

```diff
@@ -13,57 +13,70 @@
+    """Returns a compendia job for the provided organism.
+
+    Fetch all of the experiments and compile large but normally formated Dataset.
+    """
+    data = {}
+    experiments = Experiment.objects.filter(id__in=(ExperimentOrganismAssociation.objects.filter(organism=organism)).values('experiment'))
+    for experiment in experiments:
+        data[experiment.accession_code] = list(experiment.samples.filter(organism=organism).values_list('accession_code', flat=True))
+
+    job = ProcessorJob()
+    job.pipeline_applied = "COMPENDIA"
+    job.save()
+
+    dset = Dataset()
+    dset.data = data
+    dset.scale_by = 'NONE'
+    dset.aggregate_by = 'SPECIES'
+    dset.quantile_normalize = False
+    dset.save()
+
+    pjda = ProcessorJobDatasetAssociation()
+    pjda.processor_job = job
+    pjda.dataset = dset
+    pjda.save()
+
+    return job
+
+
 class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument(
-            "--organism",
+            "--organisms",
             type=str,
-            help=("Name of organism"))
+            help=("Comma separated list of organism names."))
 
     def handle(self, *args, **options):
-        """ For every (or a supplied) organism, fetch all of the experiments and compile large but normally formated Dataset.
+        """Create a compendium for one or more organisms.
 
-        Send all of them to the Smasher. Smash them. Retrieve manually as desired.
+        If --organism is supplied will immediately create a compedium
+        for it. If not a new job will be dispatched for each organism
+        with enough microarray samples.
         """
-
-        dataset_ids = []
-
-        if options["organism"] is None:
+        if options["organisms"] is None:
             all_organisms = Organism.objects.all()
         else:
-            all_organisms = [Organism.get_object_for_name(options["organism"].upper())]
-
-        for organism in all_organisms:
-            data = {}
-            experiments = Experiment.objects.filter(id__in=(ExperimentOrganismAssociation.objects.filter(organism=organism)).values('experiment'))
-            for experiment in experiments:
-                data[experiment.accession_code] = list(experiment.samples.filter(organism=organism).values_list('accession_code', flat=True))
-
-            job = ProcessorJob()
-            job.pipeline_applied = "COMPENDIA"
-            job.save()
-
-            dset = Dataset()
-            dset.data = data
-            dset.scale_by = 'NONE'
-            dset.aggregate_by = 'SPECIES'
-            dset.quantile_normalize = False
-            dset.save()
-
-            pjda = ProcessorJobDatasetAssociation()
-            pjda.processor_job = job
-            pjda.dataset = dset
-            pjda.save()
-
-            final_context = create_compendia.create_compendia(job.id)
+            organisms = options["organisms"].upper().replace(" ", "_").split(",")
```

Comment:
we probably don't need to do `replace(" ", "_")`, I think the command parameters can't have spaces

Reply:
Why not? `./workers/run_command.sh -i compendia "python3 manage.py create_compendia --organisms 'DANIO_RERIO,HOMO SAPIENS'"` works

Reply:
nice! I think I was calling the commands differently when I had the one with the regular expression